packages/harmony/src/components/button/FilterButton/FilterButton.tsx

```diff
@@ -156,9 +147,13 @@ export const FilterButton = forwardRef<HTMLButtonElement, FilterButtonProps>(
       }
     }, [selection, variant, setIsOpen, setSelection])
 
-    const handleOptionSelect = useCallback((option: FilterButtonOption) => {
-      setSelection(option)
-    }, [])
+    const handleOptionSelect = useCallback(
+      (option: FilterButtonOption) => {
+        setSelection(option.value)
+        onSelect?.(option.value)
```

Comment:
nit: It might be nicer to have the full option returned. Could be some scenarios where you want to use the label text on change and its much easier to just drill into `option.value` than re-finding it in the original array. Loosely held opinion tho

Reply:
I think I'll leave it for now, but open to having it change in the future if needed!